src/x/query.rs

```diff
@@ -10,6 +10,41 @@ use std::fmt;
 pub trait Query<X: XConn> {
     /// Run this query for a given window ID.
     fn run(&self, id: Xid, x: &X) -> Result<bool>;
+
+    /// Combine this query with another query using a logical AND.
```

Comment:
Can we add some brief explanations of the expected behaviour here (in terms of short circuiting etc)?

Reply:
Added an explanation